src/main/java/org/apache/maven/shared/incremental/IncrementalBuildHelper.java

```diff
@@ -129,6 +146,30 @@ public void setDirectoryScanner( DirectoryScanner directoryScanner )
         this.directoryScanner = directoryScanner;
     }
 
+    /**
+     * Get the existing DirectoryScanner used by this helper,
+     * or create new a DirectoryScanner if none is yet set.
+     * The DirectoryScanner is used for detecting changes in a directory
+     */
+    public DirectoryScanner getGeneratedSourcesDirectoryScanner()
```

Comment:
Do these methods need to be public? It looks like they can be package protected/default access

Reply:
These are copy-pasted from the getDirectoryScanner(). It isn't clear to me why that one is public. I made this one public to be consistent.